src/lib/misc.ml

```diff
@@ -189,6 +189,12 @@ let to_ocluster_submission spec =
   | `Production | `Dev -> to_obuilder_job spec |> Cluster_api.Submission.obuilder_build
   | `Docker -> to_docker_job spec |> Cluster_api.Submission.docker_build
 
+let with_error_check fn:('a -> string -> 'a) =
```

Comment:
```suggestion
let with_error_check ~(f : ('a -> string -> 'a))
```
Keeping with convention label the passed in function `f`

Reply:
This labeling is causing for compilation errors.